website/docs/cli/commands/test.mdx

```diff
@@ -1,12 +1,165 @@
+The option can be used multiple times to specify several files.
+
+* `-json` Set for test outputs to be formatted as JSON.
+
+* `-no-color` Disable color codes in the command output.
+
+* `-verbose` Print the plan or state for each test run block as it executes.
+
+## Usage examples
+
+### Flat structure
+
+**GIVEN** the following OpenTofu module structure:
+
+```commandline
+.
+├── LICENSE
+├── README.md
+├── main.tf
+├── main.tftest.hcl
+├── foo.tf
+├── foo.tftest.hcl
+├── bar.tf
+├── bar.tftest.hcl
+├── outputs.tf
+└── variables.tf
+```
+
+**WHEN** `tofu test` is executed in the project directory,
+
+**THEN** the tests defined in all three files, `main.tftest.hcl`, `foo.tftest.hcl` and `bar.tftest.hcl`, will be run.
+
+**WHEN** `tofu test -filter=foo.tftest.hcl -filter=bar.tftest.hcl` is executed in the project directory,
+
+**THEN** the tests defined in the files `foo.tftest.hcl` and `bar.tftest.hcl` will be run only.
+
+### Nested structure
+
+**GIVEN** the following OpenTofu module structure:
+
+```commandline
+.
+├── LICENSE
+├── README.md
+├── foo.tf
+├── foo.tftest.hcl
+├── bar.tf
+├── bar.tftest.hcl
+├── main.tf
+├── main.tftest.hcl
+├── tests
+│   ├── foo.tftest.hcl
+│   └── bar.tftest.hcl
+├── modules
+│   ├── module_foo
+│   │   ├── LICENSE
+│   │   ├── README.md
+│   │   ├── foo.tf
+│   │   ├── foo.tftest.hcl
+│   │   ├── main.tf
+│   │   ├── main.tftest.hcl
+│   │   ├── outputs.tf
+│   │   └── variables.tf
+│   └── module_bar
+│       ├── LICENSE
+│       ├── README.md
+│       ├── foo.tf
+│       ├── foo.tftest.hcl
+│       ├── bar.tf
+│       ├── bar.tftest.hcl
+│       ├── main.tf
+│       ├── main.tftest.hcl
+│       ├── outputs.tf
+│       └── variables.tf
+├── outputs.tf
+└── variables.tf
+```
+
+**WHEN** `tofu test` is executed in the project root directory,
+
+**THEN** the tests defined in the following files will be run (note that the files are sorted in the execution order):
+
+- bar.tftest.hcl,
+- foo.tftest.hcl,
+- main.tftest.hcl,
+- tests/bar.tftest.hcl,
+- tests/foo.tftest.hcl.
+
+**WHEN** `tofu test -test-directory=modules/module_bar` is executed in the project root directory,
+
+**THEN** the tests defined in the following files will be run (note that the files are sorted in the execution order):
+
+- bar.tftest.hcl,
+- foo.tftest.hcl,
+- main.tftest.hcl,
+- modules/module_bar/bar.tftest.hcl,
+- modules/module_bar/foo.tftest.hcl,
+- modules/module_bar/main.tftest.hcl.
+
+**WHEN** `tofu test -test-directory=modules/module_bar -filter=bar.tftest.hcl` is executed in the project root directory,
+
+**THEN** the tests defined in the file `bar.tftest.hcl` will be run only.
+
+**WHEN** `tofu test -test-directory=modules/module_bar -filter=modules/module_bar/bar.tftest.hcl` is executed in the project root directory,
+
+**THEN** the tests defined in the file `modules/module_bar/bar.tftest.hcl` will be run only.
+
+## Definition syntax
+
+TBD
+
+## Variables
```

Comment:
Here I'd probably give an example of overriding variables in a test file / run block, and explain why one might want to do that

Reply:
Thanks for your feedback. I'll add a couple of illustrations/use cases. 

Could you please expand on this

> explain why one might want to do that

For context, I find that the main focus of this docu page shall be more practice oriented than explanation oriented (leaning more towards the top half of the [diagram](https://documentation.divio.com/) below).

![image](https://github.com/opentofu/opentofu/assets/13434797/d78ab91a-fb5f-468b-bfd7-0a6d9be1de50)

 WDYT?